File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
        if (n <= 2) return n;
        
        var oneStepBefore = 2;
        var twoStepsBefore = 1;
        var allWays = 0;
        
        for (var i = 2; i < n; i++) {
            allWays = oneStepBefore + twoStepsBefore;
            twoStepsBefore = oneStepBefore;
            oneStepBefore = allWays;
        }
        return allWays;    
};

File: 0070-climbing-stairs/0070-climbing-stairs.cs
public class Solution {
    public int ClimbStairs(int n) {
        if (n <= 2) return n;
        
        int oneStepBefore = 2;
        int twoStepsBefore = 1;
        int allWays = 0;
        
        for (int i = 2; i < n; i++) {
            allWays = oneStepBefore + twoStepsBefore;
            twoStepsBefore = oneStepBefore;
            oneStepBefore = allWays;
        }
        return allWays;        
    }
}